### DISCUSSION TITLE
404/not found handling for v0.23 when used a Framework (JS)

### DISCUSSION THREAD

**Discussion by copongcopong** [pocketbase/pocketbase#5883] [Q&A] [answered]
when i have a route set for `/` e.g `routeAdd('/', 'GET', (c) => c.html(200, '<p>home</p>'))`  it always becomes the catch-all handler when visiting url that are none existent. 

Is this by design?  am I just missing this in the docs on how to handle non-existent/404 routes? 

The 404/not-found works though when I don't have a route assigned to `/`

I am running it like `$> ./pocketbase serve --indexFallback=false`

**Comment by ganigeorgiev** [comment, marked as answer]
Routes ending with a trailing slash match everything that begins with that path. If you want only `/` you can try to append `{$}` (e.g. `/{$}`).

This is already documented in details with the new v0.23.0 documentation, but you can find more information about the Go std mux pattern matching in https://pkg.go.dev/net/http#ServeMux and https://go.dev/blog/routing-enhancements.

**Reply by copongcopong** [reply]
Thank you @ganigeorgiev! Tried and it works!

**Comment by rudyrdx** [comment]
For me, mapping the static dir to the / route using `e.Router.GET("/{path...}", apis.Static(os.DirFS(publicDir), false))` and keeping the home with 
```go
e.Router.GET("/{$}", func(c *core.RequestEvent) error {
    return handlers.HomeHandler(c)
})
```
worked the way I wanted it to. The documentation mentions that routes ending with a trailing slash match everything that begins with that path, and you can append `{$}` to match only the root path.
